x-pack/platform/plugins/shared/fleet/server/tasks/sync_integrations/sync_integrations_on_remote.ts

```
@@ -180,3 +220,224 @@ export const syncIntegrationsOnRemote = async (
+      throw new IndexNotFoundError(`Index not found`);
+
+    logger.error('error', err.message);
+    throw err;
+  }
+};
+
+export const fetchAndCompareSyncedIntegrations = async (
+  esClient: ElasticsearchClient,
+  savedObjectsClient: SavedObjectsClientContract,
+  index: string,
+  logger: Logger
+) => {
+  try {
+    // find integrations on ccr index
+    const searchRes = await esClient.search<SyncIntegrationsData>({
+      index,
+      sort: [
+        {
+          'integrations.updated_at': {
+            order: 'desc',
+          },
+        },
+      ],
+    });
+    if (searchRes.hits.hits[0]?._source === undefined) {
+      return {
+        integrations: [],
+        error: `No integrations found on ${index}`,
+      };
+    }
+    const ccrIndex = searchRes.hits.hits[0]?._source;
+    const { integrations: ccrIntegrations, custom_assets: ccrCustomAssets } = ccrIndex;
+
+    // find integrations installed on remote
+    const installedIntegrations = await getPackageSavedObjects(savedObjectsClient);
+
+    if (!installedIntegrations && ccrIntegrations.length > 0) {
+      return {
+        integrations: [],
+        error: `No integrations installed on remote`,
+      };
+    }
+
+    const installedIntegrationsByName = (installedIntegrations?.saved_objects || []).reduce(
+      (acc, integration) => {
+        if (integration?.id) {
+          acc[integration.id] = integration;
+        }
+        return acc;
+      },
+      {} as Record<string, SavedObjectsFindResult<Installation>>
+    );
+    const customAssetsStatus = await compareCustomAssets(esClient, logger, ccrCustomAssets);
+    const integrationsStatus = compareIntegrations(ccrIntegrations, installedIntegrationsByName);
+    const result = {
+      ...integrationsStatus,
+      ...(customAssetsStatus && { custom_assets: customAssetsStatus }),
+    };
+
+    return result;
+  } catch (err) {
+    logger.error('error', err.message);
+    throw err;
```

same here, should we return the error in the response?